### PR TITLE
Refactor user quests

### LINKED ISSUE #41: BUG - User Can Generate First 3 Quests Multiple Times
**Describe the bug**
When we hit the endpoint `api/v1/users/2/quests?completion_status=false`, even when we already have the first 3 quests assigned, we are getting those reassigned again. It should be the first time, when the user has 0 user-quest records, that we generate the first 3 quests.

### LINKED ISSUE #42: BUG - Only 1 Quest is Returning for User
**Describe the bug**
When we hit the endpoint `api/v1/users/2/quests?completion_status=false` and a user has multiple user-quest records in the database, we are only getting `1` user-quest record back instead of the expected `3`

### PR DESCRIPTION
**Pull request by GeorgieGirl24**
### Description

Rework the resources.user_quests to allow for a list of quests instead of just one

### Closes issue(s)

closes #42 
closes #41 

### Testing

No additional tests were made to fix the issues

### Changes include
- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [ ] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [x] I ran full test suite - all tests passing

### CODE REVIEW

File: api/resources/user_quests.py [hunk 1]
```diff
@@ -13,28 +13,31 @@
...
     return {
         'data': {
-            'id': 'Null',
-            'type': 'quests',
-            'attributes': {
-                 "quests": [quest_objects]
+        'id': 'Null',
+        'type': 'quests',
+        'attributes': {
+            "quests": serializable_user_quests
```
Comment (jakeheft) [inline api/resources/user_quests.py]: We don't want this indented?

**Review by carson-jardine — APPROVED** [review]
LGTM

File: api/resources/user_quests.py [hunk 2]
```diff
@@ -13,28 +13,31 @@
...
-            'type': quest.type,
-            'name': quest.name,
-            'xp': quest.xp,
-            'encounter_req': quest.encounter_req,
-            'level': quest.level,
-            'progress': int(progress)
-        }
+    serializable_user_quests = []
+    for quest in quests:
+        serializable_user_quests.append(_serialize_user_quest(quest[1], quest[0]))
```
Comment (c-bartell) [inline api/resources/user_quests.py:18]: Great job cleaning up this method!

**Review by jakeheft — APPROVED** [review]
Aside from indentation everything looks great. Good work getting this working

**Review by c-bartell — APPROVED** [review]
It looks great!

**Pull request merged**